docs/topics/unused-return-value-checker.md

```diff
@@ -0,0 +1,173 @@
+[//]: # (title: Unused return value checker)
+
+<primary-label ref="experimental-general"/>
+
+> This feature is planned to be stabilized and improved in future Kotlin releases.
+> We would appreciate your feedback in our issue tracker [YouTrack](https://youtrack.jetbrains.com/issue/KT-12719).
+>
+{style="note"}
+
+The unused return value checker allows you to detect _ignored results_.
+These are values returned from expressions that produce something other than
+`Unit` or `Nothing` and aren't:
+
+* Stored in a variable or property.
+* Returned or thrown.
+* Passed as an argument to another function.
+* Used as a receiver in a call or safe call.
+* Checked in a condition such as `if`, `when`, or `while`.
+* Used as the last statement of a lambda.
+
+> The checker doesn't report ignored results from increment operations such as `++` and `--`.
+>
+{style="note"}
+
+You can use the unused return value checker to catch bugs where a function call produces a meaningful result, but the result is silently dropped.
+This helps prevent unexpected behavior and makes such issues easier to track down.
+
+Here's an example:
+
+```kotlin
+fun formatGreeting(name: String): String {
+    if (name.isBlank()) return "Hello, anonymous user!"
+    if (!name.contains(' ')) {
+        // The checker reports a warning that this result is ignored
+        "Hello, " + name.replaceFirstChar(Char::titlecase) + "!"
+        // Unused return value of 'plus'.
+    }
+    val (first, last) = name.split(' ')
+    return "Hello, $first! Or should I call you Dr. $last?"
+}
+```
+
+In this example, a string is created but never used, so the checker reports it as an ignored result.
+
+## Configure the unused return value checker
+
+You can control how the compiler reports ignored results with the `-Xreturn-value-checker` compiler option.
+
+It has the following modes:
+
+* `disable` disables the unused return value checker. (Default)
+* `check` reports warnings for ignored results from [marked functions](#mark-functions-to-check-ignored-results).
+* `full` reports warnings for ignored results from all functions in your project.
```

Comment:
It is important to highlight that `full` mode also _distributes_ your code as marked, so other users of the feature will get warnings on your library or module.